---
title: "DeepLife - Program"
layout: textlay
excerpt: ""
sitemap: false
permalink: /program
---

## Program

This course will be organized in 4 blocks over 14 weeks (start: week of 26.02.2024)

1. [Basic concepts in deep learning](#Part1) – 4 weeks
2. [DL in single-cell genomics](#Part2) – 3 weeks
3. [DL in protein bioinformatics](#Part3) - 4 weeks
4. [DL in image analysis](#Part4) – 3 weeks
<br>

The course format will comprise a weekly **90-minute online lecture** and a **weekly hybrid (in-person/online) practical Python session**. Lectures will be given by [teachers]({{ site.url }}{{ site.baseurl }}/team) from all participating universities. Lectures and practical exercises on all three application areas will be centered around one **recent publication** illustrating a specific application and method.
<br>

The course will end with a 2-day workshop and [hackathon meeting](#Hackathon) in Heidelberg in June 2024 during which students will be able to implement a short project and listen to scientific lectures.


## Prerequisites {#Prerequisites}

Students attending this course are expected to have some basic statistics knowledge and machine-learning fundamentals. 
Recommended books are among others:

1. [Deep Learning book](https://www.deeplearningbook.org/) by Goodfellow, Bengio, Courville
2. [The Elements of Statistical Learning](https://hastie.su.domains/Papers/ESLII.pdf) by Hastie, Tibshirani, Friedman
3. [An Introduction to Statistical Learning](https://www.statlearning.com/) by Hastie, Tibshirani, Friedman (a simpler version of the previous book)

Specifically, we expect that the following concepts are familiar:

#### basic statistics 

* accuracy
* sensitivity/specificity
* area under the curve (AUC)
* probability distributions
* random variable
* expectation of a random variable

#### machine-learning

* overfitting vs. underfitting
* cross-validation
* usage of training, validation and testing datasets
* classification vs. regression (supervised vs. unsupervised)
* binary vs. multi-class classification

#### mathematical foundations

* matrix algebra


## Schedule of lectures


[please as a table format here!]

| Date | Title | Lecturer | Content | Link to slides | Link to practical | 
|----|----|----|----|----|----|



## 1 - Basic concepts in deep learning {#Part1}

### 1.1 - Introduction and mathematical foundation to DL
> 26.02.2024 - 01.03.2024

#### Lecturer
- Bartek Wilczynski (Warsaw)

#### Content
- Representing functions as NNs
- preception and non-linear activation
- multi-layered networks
- back-propagation

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publication: [Name of the paper](link to the paper)

### 1.2 -  Convolutional Neural Network (CNN) and Recurrent Neural Network (RNN)
> 04.03.2024 - 08.03.2024

#### Lecturer 
- Marco Frasca (Milano)

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publication: [Name of the paper](link to the paper)

### 1.3 - Autoencoder (AE) and Variational Autoencoder (VAE) 
> 11.03.2024 - 15.03.2024

#### Lecturer 
- Carl Herrmann (Heidelberg)

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publication: [Name of the paper](link to the paper)

### 1.4 - Transformers and attention mechanism
> 18.03.2024 - 22.03.2024

#### Lecturer 
- Dario Malchiodi (Milano)

#### Content 
- attention in Deep learning
- NLP
- RNN
- transformers, transformers architecture

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publication: [Name of the paper](link to the paper)

## 2 - DL in single-cell genomics {#Part2}

### 2.1 - Transformers/RNN models for sequence analysis
> 08.04.2024 - 12.04.2024

#### Lecturer 
- Dario Malchiodi (Milano)

#### Content 
- Literature models for protein generation

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: 
    - [Madani, A., Krause, B., Greene, E.R., Subramanian, S., Mohr, B.P., Holton, J.M., Olmos, J.L., Xiong, C., Sun, Z.Z., Socher, R., et al. (2023). Large language models generate functional protein sequences across diverse families. Nature Biotechnology 41, 1099-1106.](https://doi.org/10.1038/s41587-022-01618-2)
    - [Brandes, N., Ofer, D., Peleg, Y., Rappoport, N., and Linial, M. (2022). ProteinBERT: a universal deep-learning model of protein sequence and function. Bioinformatics 38, 2102-2110.](https://doi.org/10.1093/bioinformatics/btac020)

### 2.2 -  Models for multimodal data integration
> 15.04.2024 - 19.04.2024

#### Lecturer 
- Britta Velten (Heidelberg)

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: 
    - [Lopez, R., Regier, J., Cole, M.B., Jordan, M.I., and Yosef, N. (2018). Deep generative modeling for single-cell transcriptomics. Nature Methods 15, 1053-1058.](https://doi.org/10.1038/s41592-018-0229-2)
    - [Xu, C., Lopez, R., Mehlman, E., Regier, J., Jordan, M.I., and Yosef, N. (2021). Probabilistic harmonization and annotation of single-cell transcriptomics data with deep generative models. Molecular Systems Biology 17, e9620.]( https://doi.org/10.15252/msb.20209620)
    
### 2.3 - VAE in single-cell genomics 
> 22.04.2024 - 26.04.2024

#### Lecturer 
- Carl Herrmann (Heidelberg)

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)


## 3 - DL in protein bioinformatics {#Part3}

### 3.1 - AF, EMSFold to predict structure of proteins
> 29.04.2024 - 03.05.2024

#### Lecturer 
- Joanna Sulkowska (Warsaw)

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)

### 3.2 -  RNN, CNN models for topology/graph analysis in biopolymers
> 06.05.2024 - 10.05.2024

#### Lecturer 
- ...

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)


### 3.3 - Deep learning models for protein-ligand binding site prediction 
> 13.05.2024 - 17.05.2024

#### Lecturer 
- ...

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)

### 3.4 - Diffusion models for protein design
>  20.05.2024 - 24.05.2024

#### Lecturer 
- ...

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)

### (3.5 - Deep Architectures for sampling macromolecules)
> 

#### Lecturer 
- Sergeant-Perthuis (Paris-Sorbonne)

#### Content 
- ...

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)

## 4 - DL in image analysis {#Part4}

### 4.1 - Introduction to bioimage analysis and deep learning utilization
> 27.05.2024 - 31.05.2024

#### Lecturer 
- ...

#### Content 
- Challenges of Bioimage Analysis
- Community-Driven resource for accessible deep learning
- Hands on knowledge with Noise2Void and Stardist

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications:
  - [Ouyang, W., Beuttenmueller, F., Gómez-de-Mariscal, E., Pape, C., Burke, T., Garcia-López-de-Haro, C., Russell, C., Moya-Sans, L., de-la-Torre-Gutiérrez, C., Schmidt, D., et al. (2022). BioImage Model Zoo: A Community-Driven Resource for Accessible Deep Learning in BioImage Analysis. bioRxiv](https://doi.org/10.1101/2022.06.07.495102)
  - [Gómez-de-Mariscal, E., García-López-de-Haro, C., Ouyang, W., Donati, L., Lundberg, E., Unser, M., Muñoz-Barrutia, A., and Sage, D. (2021). DeepImageJ: A user-friendly environment to run deep learning models in ImageJ. Nature Methods 18, 1192-1195.](https://doi.org/10.1038/s41592-021-01262-9)
  - [Krull, A., Buchholz, T.-O., and Jug, F. (2019). Noise2void-learning denoising from single noisy images. pp. 2129-2137.](https://doi.org/10.48550/arXiv.1811.10980)
  - [Stevens, M., Nanou, A., Terstappen, L., Driemel, C., Stoecklein, N.H., and Coumans, F.A.W. (2022). StarDist Image Segmentation Improves Circulating Tumor Cell Detection. Cancers (Basel) 14.](doi: 10.3390/cancers14122916)
  
### 4.2 -  
> 03.06.2024 - 07.06.2024

### 4.3 - Deep learning for segmentation
> 10.06.2024 - 15.06.2024


#### Lecturer 
- Karl Rohr (Heidelberg)

#### Content 
- CNNs for image analysis
- U-Net
- CellPose

#### Lecture Material
- Lecture: [Youtube/Zoom](exact link to Youtube/Zoom)
- Practical session: [Github](exact link to Github)
- Related publications: [Name of the paper](link to the paper)

## Hackathon Meeting {#Hackathon}

The course will end with a 2-day workshop and hackathon meeting in Heidelberg in **June 2024** during which students will be able to implement a short project and listen to scientific lectures.

Feel free to post your project ideas [here](https://github.com/orgs/deeplife4eu/discussions/categories/ideas-for-hackathon-projects)! 

<br>
<br>
